tests/testsuite/resolve.rs

```diff
@@ -320,6 +359,48 @@ fn test_resolving_maximum_version_with_transitive_deps() {
     assert_that(&res, is_not(contains(names(&[("util", "1.1.1")]))));
 }
 
+#[test]
+fn test_resolving_minimum_version_with_transitive_deps() {
```

Comment:
I really like this test, it concisely demonstrates both properties of  `minimal-versions`, that it chooses the smallest version while meeting all the requirements. Mabey a comment at the beginning explaining that for the next reader. @alexcrichton has been trying to teach me to leave more comments.

Also @alexcrichton, do you think this needs a full integration test that calls `cargo -Z minimal-versions` or is this sufficient?

Reply:
Oh sorry I missed this. Yeah @klausi mind adding a small integration test to ensure that the flag is plumbed to the right location?

Reply:
done!